src/ring_theory/ideal/norm.lean

```diff
@@ -408,3 +411,83 @@ lemma prime_of_irreducible_abs_norm_span {a : S} (ha : a ≠ 0)
 (ideal.span_singleton_prime ha).mp (is_prime_of_irreducible_abs_norm hI)
 
 end ideal
+
+end abs_norm
+
+section span_norm
+
+namespace ideal
+
+open submodule
+
+variables (R : Type*) [comm_ring R] {S : Type*} [comm_ring S] [algebra R S]
+
+/-- `ideal.span_norm R (I : ideal S)` is the ideal generated by mapping `algebra.norm R` over `I`.
+-/
+def span_norm (I : ideal S) : ideal R :=
+ideal.span (algebra.norm R '' (I : set S))
+
+@[simp] lemma span_norm_bot
```

Comment:
This is false without `nontrivial S`, right?

Reply:
Yes, then this would be `ideal.span {algebra.norm 1} = ideal.span {1} = ⊤` (since `norm`/`det` breaks the tie between `0 = 1` in favour of 1).